templates/item/preview.macros.njk

```diff
@@ -235,66 +229,102 @@
-  {% call show.item_section('ratio-preview', 'Ratio Previews') %}
-    {% for name, value in data %}
-      <div data-sassdoc="ratio-preview">
-        {{ annotation.param_list(
-          name=name + ':',
-          value=value
-        ) }}
-
-        {% set counts = {
-          '1': 1,
-          '2': 2,
-          '3': 3,
-          '4': 4,
-          '5': 5,
-          '6': 6,
-          '7': 7,
-          '8': 8,
-          '9': 9,
-          '10': 10
-        } %}
-
-        <div data-sassdoc="ratio-demos">
-          {% set count = counts[args.count] or 6 %}
-          {% for demo in range(count) %}
-            {% set demo = 1 * (value ** demo) if (value != 'linear') else (demo + 1) %}
-            <span data-sassdoc="ratio-demo" style="font-size: {{ demo + 'em' }};">
-              {{ demo|round(2) }}
-            </span>
-          {% endfor %}
-        </div>
-      </div>
-    {% endfor %}
-  {% endcall %}
+{% macro ratios(data) %}
+  {% for name, value in data %}
+    {% call show.item_section(
+      slug='ratio-preview',
+      title='Ratio',
+      supplement=name + ' (' + value + ')'
+    ) %}
+      {{ size_ruler(data, is_ratio=true) }}
+    {% endcall %}
+  {% endfor %}
 {% endmacro %}
 
 
 {% macro sizes(data, args) %}
   {% call show.item_section('size-preview', 'Size Previews') %}
-    {% for name, value in data %}
-      <div data-sassdoc="size-preview">
-        {{ annotation.param_list(
-          name=name + ':',
-          value=value
-        ) }}
-
-        {% set style = args.style or 'text' %}
-        {% if style != 'output-only' %}
-          <div data-size-demo="{{ style }}" style="font-size: {{ value }};">
-            {% if style == 'text' %}
-              {{ lorem('title') }}
-            {% endif %}
-          </div>
-        {% endif %}
-      </div>
-    {% endfor %}
+    {% set style = args.style or 'text' %}
+
+    {% if style == 'ruler' %}
+      {{ size_ruler(data) }}
+    {% else %}
+      {{ size_output(data, style) }}
+    {% endif %}
   {% endcall %}
 {% endmacro %}
 
 
+{% macro size_ruler(data, is_ratio=false) %}
+  <table class="rulers">
+    <tr class="is-hidden">
+      <th class="ruler-name">{{ 'Step' if is_ratio else 'Name' }}</th>
+      <th class="ruler-value">Value</th>
+      <td class="ruler-viz"></td>
+    </tr>
+
+    {% if is_ratio %}
+      {% set start = 5 %}
+      {% set count = range(1, 7) if (value == 'linear') else range(-3, 4) %}
+      {% for step in count %}
+        {% set result = (1 * step) if (value == 'linear') else (1 * (value ** step)) %}
+        {% set width = (start * step) if (value == 'linear') else (start * (value ** step)) %}
+        {% set width = width + 'em' %}
+
+        {{ size_ruler_row(
+          name='—' if ( result == 1 ) else step,
+          value='base' if ( result == 1 ) else result|round(2),
+          width=width
+          ) }}
+      {% endfor %}
+    {% else %}
+      {% for value, items in data|list|groupby('value') %}
```

Comment:
@mirisuzanne I tried sorting here (see https://github.com/oddbird/sassdoc-theme-herman/commit/159659262982ce9dca73bbb4120e75addc449179), but I think it's unhelpful unless we intelligently sort by calculated size. The simple `|sort(attribute='value')` is too dumb, and just sorts alphabetically. I'd say better to allow the user to sort manually in their Sass map.

Reply:
@jgerigmeyer I agree. That's what I saw last night and just assumed I was doing something wrong. I'll rip it out.

Reply:
Oh, you already did. Great.